### DISCUSSION TITLE
Option to show nodes of the both sides of the view root node after jump-in

### DISCUSSION THREAD

**Discussion by dpolivaev** [freeplane/freeplane#503] [Ideas]
@eu2021 suggests that after selecting a node as a view root it should be possible that its child nodes are shown on its both sides.
How do you see it? Who else needs it?

> 
> This feature would improve, at least, one of my usecases of Freeplane, which can be described like this:
> 
> I usually use my map growing only to the right side, but, for some kinds of information (eg, debates mapping), I find it very useful to distribute de information on both sides, because it gets easier to visualise and also memorize. Arguments on a debate get distributed on two fields (pro and against), so it gets visually clear when they grow to different sides around a central node.
> 
> So, I like to store the information inside my big map, with everything growing to the right side, but I like to visualize some topics distributed to both sides.
> 
> I tried some workflows to make it work (eg, exporting the topic branch, just to visualize it), but it's too cumbersome (eg, it's difficult to edit nodes when I spot mistakes while reading the exported map), so I end up not using it as much as I would like.

**Comment by Illegit** [comment]
I see it as interesting especially after your developments of jump in jump out, like having many small roots in the children only point  I'll prefer for the user is to choose if moving children one side (left or right or both), it could be default but changeable values.

I feel the use detailed by eu2021 is very effective during meetings... 95% of time, much easier than icons only.

**Comment by euu2021** [comment]
I would like to add that the original idea was from @ianikanov. In #455 he suggests submaps with left side expansion:

> [...] the drawback is that when you want to narrow down the focus and work only on specific project **you are only limited with one side expansion (further to the right)** [...]
So for example above each project may be transformed into a virtual sub-map like this:
**Left side**:
[...]

And, to be fair, this idea was already brought up by other users, like in [this request]( https://sourceforge.net/p/freeplane/featurerequests/738/), and something like that in [this thread](https://sourceforge.net/p/freeplane/discussion/758437/thread/bdf5bc09/).

About users getting confused with nodes changing position after the jump in, I think the risk is minimal, because the nodes will only change sides if the user explicitly set it to the opposite side, beforehand, right?

Another thought to add: one of the strong suits of mindmapping in comparison with simply outlining is the richer spatial distribution of the information. The problem is that, after level 2, you can't make use of 2 sides distributions. Allowing 2 side distributions on submaps would allow a richer presentation of the information, even on deeper levels of the map.

**Comment by macmarrum** [comment]
For me the target solution would be for Freeplane to support a left-side position on any level, not limited to children of root. Taking it further, it would be even better for any node to grow not only to the left or right, but also up or down (as mentioned in #501).

Keeping in mind technical difficulties to achieve new automatic layouts, I'll be happy with the ability to have left-side nodes for a jumped-in root. 
Additionally, it would be great if the jumped-in side was remembered, so that after jumping out and back in, the previously-jumped-in-left nodes moved to the left automatically.

**Comment by euu2021** [comment]
Sometimes I read the OutlinerSoftware forum, and I just saw that, some days ago, there was a post there that is relevant to this thread. An experienced user said: 

> [...]
There are three shortcomings to most mind map tools for me.
Most mind maps won’t let you rotate the map to a new root node.
[...]

(it's in the last posts of [this thread](https://www.outlinersoftware.com/topics/viewt/9786/25))

Interestingly, in the sequence, another user (not me) suggests Freeplane for that user.

**Comment by EdoFro** [comment]
Hi,
I like the idea of having the option to show nodes on the both sides of the view root node after jump-in.

And here is how I think it should work:
- only child nodes of the "view root" node have the possibility to be changed from one side to another.
- this preference get saved in the node as a property
  - its descendants don't get their property changed (saved)
- After multiple Jumps in and out and having changed the node orientation on multiple levels, there will be multiple nodes with this property different as in a "normal" map. Freeplane should use following logic to build the view:
  - if the node is not a childnode of the view root node, that preference is ignored by Freeplane and it is shown as a normal child (takes the orientation of the branch)
- nodes that are part of a summary node that are side changed are no longer part of the summary (same behavior than when this happens with the root node en its children)
- nodes that are part of a cloud: it doesn't matter: to be part of a cloud it depends on the node's parent and a node that changes orientation doesn't change its parent.
- if the selected node to jump in has a cloud, then the cloud is not showed in the jump in view, so it doesn't get messed if a child changes side.

These are my humble ideas. I hope they help.
bye,
edo

**Comment by wordmuse** [comment]
It works for me, but I'm not sure I have a use for it just now. However, since I now see what is possible, that may change. Reminds me of hoisting in other outliner software.

~ Bal
